packages/lesswrong/server/repos/BookmarksRepo.ts

```diff
@@ -0,0 +1,70 @@
+import { recordPerfMetrics } from "./perfMetricWrapper";
+import AbstractRepo from "./AbstractRepo";
+import { Bookmarks } from "../collections/bookmarks/collection";
+
+interface UltraFeedBookmark {
+  documentId: string;
+  collectionName: string;
+  postId?: string;
+  directChildrenCount?: number;
+}
+
+class BookmarksRepo extends AbstractRepo<"Bookmarks"> {
+  constructor() {
+    super(Bookmarks);
+  }
+
+  /**
+   * Gets bookmark items for the UltraFeed
+   * Prioritizes more recent bookmarks with some randomness.
+   */
+  public async getBookmarksForFeed(
+    context: ResolverContext,
+    limit = 10
+  ): Promise<UltraFeedBookmark[]> {
+    const db = this.getRawDb();
```

Comment:
I'd prefer to keep this inlined, since that's how we do it everywhere else in the codebase and I was briefly confused as to how you were fetching records that weren't single-collection-shaped directly using `manyOrNone` (which should probably just be `any`).

Reply:
(we don't do it everywhere else/I haven't been, but ok!)